localstack/utils/docker.py

```diff
@@ -0,0 +1,401 @@
+        """Removes container with given name"""
+        cmd = self._docker_cmd() + ["rm"]
+        if force:
+            cmd.append("-f")
+        cmd.append(container_name)
+        LOG.debug("Removing container with cmd %s", cmd)
+        try:
+            safe_run(cmd)
+        except subprocess.CalledProcessError as e:
+            if "No such container" in e.stdout.decode(config.DEFAULT_ENCODING):
+                raise NoSuchContainer(
+                    "Docker container not found", container_name, e.stdout, e.stderr
+                )
+            else:
+                raise ContainerException(
+                    "Docker process returned with errorcode %s" % e.returncode, e.stdout, e.stderr
+                )
+
+    def list_containers(self, filter: Union[List[str], str, None] = None) -> List[dict]:
+        """List all containers matching the given filters
+
+        Returns a list of dicts with keys id, image, name, labels, status
+        """
+        filter = [filter] if isinstance(filter, str) else filter
+        cmd = self._docker_cmd()
+        cmd += ["ps", "-a"]
+        options = []
+        if filter:
+            options += [y for filter_item in filter for y in ["--filter", filter_item]]
+        cmd += options
+        cmd.append("--format")
+        cmd.append(
+            '{"id":"{{ .ID }}","image":"{{ .Image }}","name":"{{ .Names }}",'
+            '"labels":"{{ .Labels }}","status":"{{ .State }}"}'
+        )
+        LOG.debug(cmd)
+        try:
+            cmd_result = safe_run(cmd).strip()
+        except subprocess.CalledProcessError as e:
+            raise ContainerException(
+                "Docker process returned with errorcode %s" % e.returncode, e.stdout, e.stderr
+            )
+        container_list = []
+        if cmd_result:
+            container_list = [json.loads(line) for line in cmd_result.splitlines()]
+        return container_list
+
+    def copy_into_container(self, container_name: str, local_path: str, container_path: str):
+        """Copy contents of the given local path into the container"""
+        cmd = self._docker_cmd()
+        cmd += ["cp", local_path, f"{container_name}:{container_path}"]
+        LOG.debug(cmd)
+        safe_run(cmd)
+
+    def get_container_entrypoint(self, docker_image: str) -> str:
+        """Get the entry point for the given image"""
+        LOG.debug("Getting the entrypoint for image: %s", docker_image)
+        cmd = self._docker_cmd()
+        cmd += [
+            "image",
+            "inspect",
+            '--format="{{ .Config.Entrypoint }}"',
+            docker_image,
+        ]
+
+        LOG.debug(cmd)
+        try:
+            run_result = safe_run(cmd)
+        except subprocess.CalledProcessError as e:
+            if "No such image" in e.stdout.decode(config.DEFAULT_ENCODING):
+                raise NoSuchImage("Image not found", docker_image, e.stdout, e.stderr)
```

Comment:
See comment above - `message` argument ("Image not found") seems a bit redundant here.